cinema/permissions.py

```diff
@@ -0,0 +1,13 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+from rest_framework.request import Request
+from rest_framework.views import View
+from typing import Any
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request: Request, view: View) -> bool:
+        if request.method in SAFE_METHODS:
+            return request.user and request.user.is_authenticated
+        if view.basename == "order":
+            return request.user and request.user.is_authenticated
```

Comment:
The logic here checks if the view's basename is 'order', and if so, it returns whether the user is authenticated. This logic might be redundant or unclear, as it duplicates the check for authenticated users in `SAFE_METHODS`. Consider revising this logic to ensure clarity and avoid redundancy.